packages/mobile/src/account/profileInfo.ts

```diff
@@ -0,0 +1,323 @@
+    const dataURL = getDataURL(mimeType, data)
+    const pictureAccessor = new PrivatePictureAccessor(offchainWrapper)
+    writeError = yield call([pictureAccessor, 'write'], dataURL, [])
+    if (writeError) {
+      Logger.error(TAG + '@uploadNameAndPicture', writeError)
+      throw Error('Unable to write data')
+    }
+    Logger.info(TAG + 'uploadNameAndPicture', 'uploaded profile picture')
+  }
+}
+
+// this function gives permission to the recipient to view the user's profile info
+export function* uploadSymmetricKeys(recipientAddresses: string[]) {
+  // TODO: check if key for user already exists, skip if yes
+  const account = yield select(currentAccountSelector)
+  const contractKit = yield call(getContractKit)
+  yield call(unlockDEK)
+  const offchainWrapper = new UploadServiceDataWrapper(contractKit, account)
+
+  const nameAccessor = new PrivateNameAccessor(offchainWrapper)
+  let writeError = yield call([nameAccessor, 'allowAccess'], recipientAddresses)
+
+  const pictureUri = yield select(pictureSelector)
+  if (pictureUri) {
+    const pictureAccessor = new PrivatePictureAccessor(offchainWrapper)
+    writeError = yield call([pictureAccessor, 'allowAccess'], recipientAddresses)
+  }
+
+  Logger.info(TAG + '@uploadSymmetricKeys', 'uploaded symmetric keys for ' + recipientAddresses)
+
+  if (writeError) {
+    Logger.error(TAG + '@uploadSymmetricKeys', writeError)
+    throw Error('Unable to write keys')
+  }
+}
+
+export function* getProfileInfo(address: string) {
+  // TODO: check if we already have profile info of address
+  const account = yield select(currentAccountSelector)
+  const contractKit = yield call(getContractKit)
+  yield call(unlockDEK)
+  const offchainWrapper = new UploadServiceDataWrapper(contractKit, account)
+
+  const nameAccessor = new PrivateNameAccessor(offchainWrapper)
+  try {
+    const name = yield call([nameAccessor, 'read'], address)
+
+    const pictureAccessor = new PrivatePictureAccessor(offchainWrapper)
+    let picturePath
+    try {
+      const pictureObject = yield call([pictureAccessor, 'read'], address)
+      const pictureURL = pictureObject.toString()
+      picturePath = yield call(saveRecipientPicture, pictureURL, address)
+    } catch (error) {
+      Logger.warn("can't fetch picture for", address)
+    }
+    return { name: name.name, thumbnailPath: picturePath }
+  } catch (error) {
+    Logger.warn("can't fetch name for", address)
+  }
+}
+
+function* unlockDEK(addAccount = false) {
+  const privateDataKey: string | null = yield select(dataEncryptionKeySelector)
+  if (!privateDataKey) {
+    throw new Error('No data key in store. Should never happen.')
+  }
+  const dataKeyaddress = normalizeAddressWith0x(
+    privateKeyToAddress(ensureLeading0x(privateDataKey))
+  )
+  const wallet: UnlockableWallet = yield call(getWallet)
+  if (addAccount) {
+    try {
+      yield call([wallet, wallet.addAccount], privateDataKey, '')
```

Comment:
Though currently the data encryption key was stored in an unprotected way in the redux state.
We now have the possibility to secure it here and should not use an empty passphrase.
It seems we could protect it the same way we protect the main account key (PIN + generated password).
Thoughts @nategraf @nambrot @alex-m24 @cmcewen?

Reply:
Absolutely agree that Valora should not be storing the keys unprotected in redux

Reply:
I'd agree we should add some protection if possible. In order to support good UX, we should make sure it's still accessible without the PIN though because it's used so often. (We also would not keep the PIN in memory if possible, but that one is a bit more debatable given how low-entropy PINs are anyway....)

Reply:
I also don't think we should require a PIN for the DEK, it would be really confusing for the user to remember two PINs and when to input which one. Is it possible to add a password to the DEK without a PIN?

Reply:
Issue for removing DEK from redux: https://github.com/celo-org/wallet/issues/17